transport/tchannel/config.go

```diff
@@ -92,26 +91,39 @@ func (ts *transportSpec) Spec(opts ...Option) config.TransportSpec {
 }
 
 func (ts *transportSpec) buildTransport(tc *TransportConfig, k *config.Kit) (transport.Transport, error) {
-	var config transportConfig
-	// Default configuration.
-	config.tracer = opentracing.GlobalTracer()
-	config.name = k.ServiceName()
-
-	// Apply the spec's transport options.
-	for _, opt := range ts.transportOptions {
-		opt(&config)
+	var cfg transportConfig
+	for _, o := range ts.transportOptions {
+		o(&cfg)
 	}
 
-	// Override options with configuration.
-	if tc.Address != "" {
-		config.addr = tc.Address
+	if cfg.name != "" {
+		return nil, fmt.Errorf("TransportSpec does not accept ServiceName")
 	}
 
-	return config.newTransport(), nil
+	if cfg.addr != "" {
+		return nil, fmt.Errorf("TransportSpec does not accept ListenAddr")
+	}
+
+	if cfg.ch != nil {
+		return nil, fmt.Errorf("TransportSpec does not accept WithChannel")
```

Comment:
Is it worth clarifying that these errors are specific to the *TChannel* TransportSpec? (I realize that the options are specific to TChannel, but if the build is config-driven, users may not realize that.)